src/dispatching/dialogue/dialogue_dispatcher.rs

```diff
@@ -33,246 +38,394 @@ pub struct DialogueDispatcher<D, S, H, Upd> {
     /// A value is the TX part of an unbounded asynchronous MPSC channel. A
     /// handler that executes updates from the same chat ID sequentially
     /// handles the RX part.
-    senders: Arc<Map<i64, mpsc::UnboundedSender<UpdateWithCx<Upd>>>>,
+    senders: Arc<Map<i64, mpsc::UnboundedSender<Update>>>,
```

Comment:
With https://github.com/teloxide/teloxide/issues/324 in mind: maybe it's better to use [`evmap`](https://docs.rs/evmap/10.0.2/evmap/)? I haven't carefully looked at the code yet, but it seems that `evmap` would fit well here.